frontend/src/App.vue

```diff
@@ -0,0 +1,176 @@
+    x: number;
+    y: number;
+    width: number;
+    height: number;
+  };
+}
+
+interface IShapeRect extends IShapeBase {
+  kind: 'rect';
+}
+
+interface IShapeCircle extends IShapeBase {
+  kind: 'circle';
+  center: {
+    x: number;
+    y: number;
+  },
+  radius: number;
+}
+
+type IShape = IShapeRect | IShapeCircle;
+
+const offsetX = ref(0.0);
+const offsetY = ref(0.0);
+
+const mouseScreenX = ref(0.0);
+const mouseScreenY = ref(0.0);
+
+const windowWidth = ref(window.innerWidth);
+const windowHeight = ref(window.innerHeight);
+
+const viewBox = computed(() => {
+  return `${offsetX.value} ${offsetY.value} ${windowWidth.value} ${windowHeight.value}`;
+});
+
+const cursorStyle = computed(() => {
+  return `top: ${mouseScreenY.value}px; left: ${mouseScreenX.value}px;`;
+});
+
+const shapes = reactive<IShape[]>([]);
+
+const onWindowResized = () => {
+  windowWidth.value = window.innerWidth;
+  windowHeight.value = window.innerHeight;
+};
+
+const screenToBoardCoordinates = (pos: { x: number, y: number }) => {
+  return { x: pos.x + offsetX.value, y: pos.y + offsetY.value };
+};
+
+const boardToScreenCoordinates = (pos: { x: number, y: number }) => {
+  return { x: pos.x - offsetX.value, y: pos.y - offsetY.value };
+};
+
+const isDragging = ref(false);
+
+const onMouseDown = (event: MouseEvent) => {
+  switch (event.button) {
+    case 0: // left click
+    {
+      const center = screenToBoardCoordinates({ x: event.clientX, y: event.clientY });
+      shapes.push({
+        id: nanoid(),
+        kind: 'circle',
+        boundingBox: {
+          x: center.x - 20,
+          y: center.y - 20,
+          width: 40,
+          height: 40,
+        },
+        center,
+        radius: 20,
+      });
+      break;
+    }
+    case 1: // middle/wheel click
+      isDragging.value = true;
+      break;
+    case 2: // right click
+      break;
+    default:
+      break;
+  }
+};
+
+const onMouseUp = () => {
+  isDragging.value = false;
+};
+
+const onMouseMove = (event: MouseEvent) => {
+  mouseScreenX.value = event.clientX;
+  mouseScreenY.value = event.clientY;
+
+  if (isDragging.value) {
+    offsetX.value -= event.movementX;
+    offsetY.value -= event.movementY;
+  }
+};
+
+onMounted(() => {
+  window.addEventListener('resize', onWindowResized);
+  window.addEventListener('mouseup', onMouseUp);
+  window.addEventListener('mousedown', onMouseDown);
+  window.addEventListener('mousemove', onMouseMove);
+});
+
+onUnmounted(() => {
+  window.removeEventListener('resize', onWindowResized);
+  window.removeEventListener('mouseup', onMouseUp);
+  window.removeEventListener('mousedown', onMouseDown);
+  window.removeEventListener('mousemove', onMouseMove);
+});
+</script>
+
+<template>
+  <svg class="cursor" :style="cursorStyle" viewBox="0 0 32 32" width="32" height="32">
+    <path d="M0,0 L32,8 L16,16 L8,32 Z" fill="black"></path>
+  </svg>
+
+  <svg :width="windowWidth.value" :height="windowHeight.value" :viewBox="viewBox" class="board">
```

Comment:
```suggestion
  <svg :width="windowWidth" :height="windowHeight" :viewBox="viewBox" class="board">
  // removed .value on windowWidth and windowHeight, vue automatically unwraps the refs
```

Reply:
What on earth lol, I don't remember adding the `.value` here. I think it wasn't in the original commit so idk what happened.

Once I wrap up work for the day I'll fix and push, or if you're free I don't mind if you fix and push to this branch